### PR TITLE
Enhancement/refactor typescript

### PR DESCRIPTION
**Pull request by herniadlf**
Related issue https://github.com/kleros/governor-snapshot/issues/10

### CODE REVIEW

**Comment by greenlucid** [comment]
Looks good, some feedback I can give through skimming through the code:

- some `.tsx` files could be `.ts` instead since they don't use JSX syntax, such as `./src/types.tsx` or `./src/constants/chains.tsx`. Making them `.ts` is cleaner (this is just a nitpick, but technically correct)
- I think we avoid using `class`, `this`, and all those OOP features as much as possible, are they really necessary? I didn't dig much, but why is [something like this](https://github.com/herniadlf/governor-snapshot/blob/3c94d81a69802f589c823ae5e3a5c182abd3bc45/src/types.tsx#L76) preferable to something like this?:

```ts
interface Session {
    currentSessionNumber: number,
    disputeID: number
}
```

why are those constructors needed?

**Comment by herniadlf** [comment]
> Looks good, some feedback I can give through skimming through the code:
> 
> * some `.tsx` files could be `.ts` instead since they don't use JSX syntax, such as `./src/types.tsx` or `./src/constants/chains.tsx`. Making them `.ts` is cleaner (this is just a nitpick, but technically correct)
> * I think we avoid using `class`, `this`, and all those OOP features as much as possible, are they really necessary? I didn't dig much, but why is [something like this](https://github.com/herniadlf/governor-snapshot/blob/3c94d81a69802f589c823ae5e3a5c182abd3bc45/src/types.tsx#L76) preferable to something like this?:
> 
> ```ts
> interface Session {
>     currentSessionNumber: number,
>     disputeID: number
> }
> ```
> 
> why are those constructors needed?

Thanks for ur feedback @greenlucid ! Both issues have to do with typescript misinterpretations. I took your feedback, did some reading and I think I've corrected it all:

- I moved to .ts those files that don't have react at all.
- I removed the classes from the types, leaving only the interfaces.

Next week i'll be testing all the changes on goerli (i think governor and kleros liquid are there) to test the appeal module. The happy path it's good.

File: src/components/appeal/appeal-side-box.tsx [hunk 1]
```diff
@@ -67,10 +67,12 @@ const AppealSideBox: React.FC<{
     (Number(p.appealFee || 0) - Number(p.amountContributed || 0)).toString()
   );
   const useOnFund = (amount: any) => {
```
Comment (greenlucid) [inline src/components/appeal/appeal-side-box.tsx:69]: What is the type of this amount supposed to be? `BigNumber`? `string`? Can you pass decimals here? This amount should be typed, is there any reason this wasn't the case?
Same for `winner` in the props of the `AppealSideBox` (is it a number? an address?)

**Review by greenlucid — CHANGES_REQUESTED** [review]

File: src/components/appeal/appeal-side-box.tsx (hunk 1 above)
Comment (herniadlf) [reply src/components/appeal/appeal-side-box.tsx]: @greenlucid Yep, I had doubts about some objects types.. But after I saw your comment I've made a review about all the "any" types and I think that it's okay now. lmk what you think

**Comment by greenlucid** [comment]
Seems like the bug that made `withdrawTransactionList` revert had been undiscovered for 3 years:
https://github.com/kleros/governor-snapshot/commit/c6d454ff31aff5907771fa40559656c41a60c73c

**Comment by greenlucid** [comment]
There's an issue I haven't been able to solve. On occasion, a transparent iframe will get in the way of the app and prevent you from interacting. If you F12 and delete it from the HTML inspector, the app works normally. Got any idea why that happens? I haven't replicated the issue in live governor.kleros.io, only on your branch. I don't know how to reproduce the bug consistently. It occurred on the following circumstances:
- using firefox
- running the app with `yarn start` (despite it being handled with npm)

If you can't reproduce it, I could just merge and solve it if it ever becomes an issue

**Comment by herniadlf** [comment]
> There's an issue I haven't been able to solve. On occasion, a transparent iframe will get in the way of the app and prevent you from interacting. If you F12 and delete it from the HTML inspector, the app works normally. Got any idea why that happens? I haven't replicated the issue in live governor.kleros.io, only on your branch. I don't know how to reproduce the bug consistently. It occurred on the following circumstances:
> 
> * using firefox
> * running the app with `yarn start` (despite it being handled with npm)
> 
> If you can't reproduce it, I could just merge and solve it if it ever becomes an issue

I can reproduce it in master also. It's related to [hot reload in local environment](https://github.com/facebook/create-react-app/issues/11880#issuecomment-1005409614). I fixed it following [this answer](https://github.com/facebook/create-react-app/issues/11880#issuecomment-1005409614).

I've also fixed:
- index class definitions 
- a warning in the poh.svg logo

**Comment by herniadlf** [comment]
@greenlucid with the new package-json updates something in the build is broken. If you are okay, I can try to solve the iframe issue as a separate PR, because it's not related with the typescript refactor

**Comment by greenlucid** [comment]
Let's just see what happens. Sure, then, if the iframe issue won't be noticed in production, then it's fine.

**Pull request merged**